client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -54,23 +63,42 @@ import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.res.loadSvgPainter
 import androidx.compose.ui.unit.Density
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
+import com.google.common.base.Preconditions.*
+import io.spine.client.EventFilter.*
+import io.spine.examples.shareaware.MoneyCalculator
+import io.spine.examples.shareaware.PurchaseId
+import io.spine.examples.shareaware.client.payment.Dialog
+import io.spine.examples.shareaware.client.wallet.PopUpMessage
+import io.spine.examples.shareaware.investment.command.PurchaseShares
+import io.spine.examples.shareaware.investment.event.SharesPurchased
 import io.spine.examples.shareaware.market.AvailableMarketShares
+import io.spine.examples.shareaware.market.rejection.Rejections.SharesCannotBeObtained
 import io.spine.examples.shareaware.server.market.MarketProcess
 import io.spine.examples.shareaware.share.Share
+import io.spine.examples.shareaware.wallet.rejection.Rejections.InsufficientFunds
 import io.spine.money.Money
 import io.spine.util.Exceptions.*
 import java.io.IOException
 import java.net.URL
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 /**
  * UI model for the `MarketPage`.
  */
-public class MarketPageModel(client: DesktopClient) {
+public class MarketPageModel(private val client: DesktopClient) {
     private val sharesSubscriptions: EntitySubscription<AvailableMarketShares> =
         EntitySubscription(AvailableMarketShares::class.java, client, MarketProcess.ID)
+    private val purchaseState: MutableStateFlow<Boolean> = MutableStateFlow(false)
```

Comment:
I think we should have at least a separate type for this operation. 

Right now it's written in a procedural style.